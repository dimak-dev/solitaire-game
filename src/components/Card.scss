.card {
    width: 100%;
    height: 100%;
    display: inline-block;
    border-radius: 1rem;
    box-shadow: 1px 1px 1px 1px black;
    //border-style: solid;
    border-width: 1px;
    background: white;
    border-color: black;
    color: black;
    font-size: 3.5vw;

    &.red {
        color: red;
    }

    .flex-wrapper {
        display: flex;
        flex-direction: column;
        height: 100%;
        width: 100%;
    }

    .top-line {
        flex: 1;
    }

    .bottom-line {
        flex: 1;
        rotate: 180deg;
    }
}
