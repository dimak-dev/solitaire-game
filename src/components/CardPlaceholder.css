.card-placeholder {
    width: 14.2vw;
    max-width: 154px;
    height: 19.5vw;
    max-height: 209px;
    display: block;
    position: relative;
    border-radius: 1rem;
    background-color: rgba(25, 137, 234, 0.3);
    overflow: hidden;
}



.card-placeholder .icon-wrapper {
    display: flex;
    align-items: center;
    justify-content: center;
    width: 100%;
    height: 100%;
}

.card-placeholder .icon-wrapper .icon {
    color: rgba(255, 255, 255, 0.499);
    padding: 5px;
    display: inline-block;
    font-size: 3vw;
}

@media (min-width: 1080px) {
    .card-placeholder .icon-wrapper .icon {
        font-size: 32px;
    }
}