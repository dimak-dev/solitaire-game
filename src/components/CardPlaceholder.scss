.card-placeholder {
    width: 14.2vw;
    max-width: 154px;
    height: 19.5vw;
    max-height: 209px;
    display: block;
    position: relative;
    border-radius: 1rem;
    border-style: solid;
    border-color: rgba(25, 137, 234);
    border-width: 1px;
    box-sizing: border-box;
    background-color: rgba(25, 137, 234, 0.3);
    overflow: hidden;

    &.target {
        background-color: rgba(25, 137, 234, 0.5);
    }

    .icon-wrapper {
        display: flex;
        align-items: center;
        justify-content: center;
        width: 100%;
        height: 100%;

        .icon {
            color: rgba(255, 255, 255, 0.499);
            padding: 5px;
            display: inline-block;
            font-size: 3vw;

            @media (min-width: 1080px) {
                font-size: 32px;
            }
        }
    }
}
