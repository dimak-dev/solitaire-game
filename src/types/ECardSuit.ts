/**
 * Enumeration of playing card suites.
 *
 * Values were previously obfuscated for decreasing cheating with devtools.
 */
export enum ECardSuit {
    DIAMOND = 'gUnw0pH2E7',
    CLUB = 'mtSk8H9gkh',
    HEART = 'sGxm9srzb2',
    SPADE = 'g6N2RYyLdf',
}