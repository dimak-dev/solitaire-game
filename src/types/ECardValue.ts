/**
 * Enumeration of playing card values.
 *
 * Values were previously obfuscated for decreasing cheating with devtools.
 */
export enum ECardValue {
    ACE = 'ry4RBhALfp',
    ONE = 'ft2meneEvA',
    TWO = 'rqqHztNSTK',
    THREE = '3Np8ndDRvY',
    FOUR = 'pbpuMgeW2K',
    FIVE = 'DEUDTax2ph',
    SIX = 'ad9CpwgFL2',
    SEVEN = 'JF42qgpdC3',
    EIGHT = 'N5XLKDTQrH',
    NINE = '7ZK2SeverJ',
    JACK = 'EEYmTfXMEx',
    QUEEN = 'bBEZ2TnDrV',
    KING = 'xmXPdGVDTn',
}