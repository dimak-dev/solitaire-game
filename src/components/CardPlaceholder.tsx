import React from 'react';
import './CardPlaceholder.css';

interface ICardPlaceholderProps {
    children?: JSX.Element,
}

export default function CardPlaceholder({children}: ICardPlaceholderProps) {
    return (
        <div className="card-placeholder">

            {children || (
                <div className='icon-wrapper'>
                    <div className='icon'>♤ ♧<br/>♡ ♢</div>
                </div>
            )}
        </div>
    )
};