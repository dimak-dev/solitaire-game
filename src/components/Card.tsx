import React, {memo} from 'react';
import './Card.scss';
import {ECardSuit} from "Types/ECardSuit";
import {ECardValue} from "Types/ECardValue";
import cardValueToChar from "Utils/cardValueToChar";
import cardSuitToChar from "Utils/cardSuitToChar";
import {cardSuiteColor} from "Utils/cardSuiteColor";


interface ICardProps {
    suit: ECardSuit,
    value: ECardValue,
}

/**
 * Card component.
 *
 * Memoized for decreasing count of renders.
 */
// const Card = memo();

export default function Card({suit, value}: ICardProps) {
    const valueChar = cardValueToChar(value);
    const suitChar = cardSuitToChar(suit);
    const color = cardSuiteColor(suit);

    return (
        <div className={`card ${color}`}>
            <div className="flex-wrapper">
                <div className="top-line">
                    {valueChar}{suitChar}
                </div>
                <div className="bottom-line">
                    {valueChar}{suitChar}
                </div>
            </div>
        </div>
    )
};