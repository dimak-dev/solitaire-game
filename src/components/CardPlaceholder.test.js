import {render, screen} from '@testing-library/react';
import renderer from 'react-test-renderer';
import Card from './Card';
import {ECardSuit} from "Types/ECardSuit";
import {ECardValue} from "Types/ECardValue";
import CardPlaceholder from "Components/CardPlaceholder";

test('Snapshot of CardPlaceholder Component', () => {
    const tree = renderer
        .create(<CardPlaceholder/>)
        .toJSON();

    expect(tree).toMatchSnapshot();
});

test('Snapshot of CardPlaceholder Component with Card as Children', () => {
    const tree = renderer
        .create(<CardPlaceholder><Card suit={ECardSuit.HEART} value={ECardValue.ONE}/></CardPlaceholder>)
        .toJSON();

    expect(tree).toMatchSnapshot();
});