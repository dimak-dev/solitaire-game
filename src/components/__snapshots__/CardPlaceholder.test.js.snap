// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Snapshot of CardPlaceholder Component 1`] = `
<div
  className="card-placeholder"
>
  <div
    className="icon-wrapper"
  >
    <div
      className="icon"
    >
      ♤ ♧
      <br />
      ♡ ♢
    </div>
  </div>
</div>
`;

exports[`Snapshot of CardPlaceholder Component with Card as Children 1`] = `
<div
  className="card-placeholder"
>
  <div
    className="card red"
  >
    <div
      className="flex-wrapper"
    >
      <div
        className="top-line"
      >
        1
        ♥
      </div>
      <div
        className="bottom-line"
      >
        1
        ♥
      </div>
    </div>
  </div>
</div>
`;
